.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,472 @@
+name: Regression Tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      hardhat-ref:
+        required: true
+        type: string
+        default: 'v-next'
+      edr-ref:
+        required: false
+        type: string
+        default: ''
+      repositories:
+        required: false
+        type: string
+        default: '["Elytro-eth/soul-wallet-contract", "foundry-rs/forge-std", "kalidao/keep", "mds1/multicall", "pancakeswap/infinity-core", "PaulRBerg/prb-math", "PaulRBerg/prb-proxy", "PaulRBerg/prb-test", "pcaversaccio/createx", "ProjectOpenSea/seaport", "sablier-labs/lockup", "sablier-labs/v2-periphery", "transmissions11/solmate", "Uniswap/UniswapX", "Vectorized/solady"]'
+      runners:
+        required: false
+        type: string
+        default: '["ubuntu-latest"]'
+      commands:
+        required: false
+        type: string
+        default: '["hardhat compile", "hardhat test solidity"]'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-edr:
+    name: Build EDR
+    if: inputs.edr-ref
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        runner: ${{ fromJSON(inputs.runners) }}
+    defaults:
+      run:
+        shell: bash
+        working-directory: crates/edr_napi
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: NomicFoundation/edr
+          ref: ${{ inputs.edr-ref }}
+      - id: target
+        env:
+          RUNNER: ${{ matrix.runner }}
+        run: |
+          if [[ "$RUNNER" == "ubuntu-latest" ]]; then
+            echo "target=x86_64-unknown-linux-gnu" >> $GITHUB_OUTPUT
+          elif [[ "$RUNNER" == "macos-latest" ]]; then
+            echo "target=aarch64-apple-darwin" >> $GITHUB_OUTPUT
+          elif [[ "$RUNNER" == "windows-latest" ]]; then
+            echo "target=x86_64-pc-windows-msvc" >> $GITHUB_OUTPUT
+          else
+            exit 1
+          fi
+      - uses: ./.github/actions/setup-node
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+      - name: Add Rust cross-compilation target
+        run: rustup target add ${{ steps.target.outputs.target }}
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile --prefer-offline
+      - name: Build
+        run: pnpm build --target ${{ steps.target.outputs.target }}
+      - name: Remove symbols
+        if: runner.os != 'MacOS'
+        run: strip *.node
+      - name: Remove symbols (macOS)
+        if: runner.os == 'MacOS'
+        run: strip -x *.node
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: edr-${{ matrix.runner }}
+          path: crates/edr_napi/edr.*.node
+          if-no-files-found: error
+
+  publish-edr:
```

Comment:
Most of the EDR publishing job was copied from https://github.com/NomicFoundation/edr/blob/main/.github/workflows/edr-npm-release.yml. Here, by publishing, we mean creating a EDR package tarball and uploading it as an artifact.